android/src/main/java/com/marigold/rnsdk/RNMarigoldPackage.kt

```diff
@@ -0,0 +1,115 @@
+package com.marigold.rnsdk
+
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import com.facebook.react.ReactPackage
+import com.facebook.react.bridge.JavaScriptModule
+import com.facebook.react.bridge.NativeModule
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.uimanager.ViewManager
+import com.marigold.sdk.Marigold
+
+class RNMarigoldPackage : ReactPackage {
+
+    var displayInAppNotifications = true
+
+    @VisibleForTesting
+    private val marigold = Marigold()
+
+    /**
+     * Default constructor - for auto linking
+     */
+    constructor() {
+    }
+
+    constructor(context: Context, appKey: String) {
```

Comment:
I think we should remove this constructor in favour of always calling `startEngine` through the native SDK as early as possible.